guide/src/module.md

```diff
@@ -1,44 +1,49 @@
 # Python Modules
 
-You can create a module as follows:
+You can create a module using `#[pymodule]`:
 
 ```rust
 use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
 
-// add bindings to the generated Python module
-// N.B: "rust2py" must be the name of the `.so` or `.pyd` file.
+#[pyfunction]
+fn double(x: usize) -> usize {
+    x * 2
+}
 
 /// This module is implemented in Rust.
 #[pymodule]
-fn rust2py(py: Python, m: &PyModule) -> PyResult<()> {
-    // PyO3 aware function. All of our Python interfaces could be declared in a separate module.
-    // Note that the `#[pyfn()]` annotation automatically converts the arguments from
-    // Python objects to Rust values, and the Rust return value back into a Python object.
-    // The `_py` argument represents that we're holding the GIL.
-    #[pyfn(m)]
```

Comment:
I personally prefer the `#[pyfunction]` approach and think it makes things easier to understand. Better separation of concerns, such that the `#[pymodule]` function is solely for FFI registration and has no business logic.

I think the example is easier to follow by using `#[pyfunction]` - `function.md` still demos `#[pyfn]`.

Reply:
That's great! I was going to open an issue proposing that very thing :) I think there's value in having a consolidate way to do things. 

Would you like me to open an issue for this? I'd be happy to implement.

Reply:
Yeah, I don't like having two slightly different ways to do the same thing.

As for an issue, there is one: #694

Reply:
Completely agree that `#[pyfn]` should go. The syntax will at least be _almost_ the same in PyO3 0.14.

Regarding #694 - the vision was that rather than remove `#[pyfn]`, we would first create a new `#[pymodule]` syntax which is used on `mod my_module` instead of `fn my_module`. Then we could deprecate the `#[pymodule]` function syntax and `#[pyfn]` together.